src/services/state-persistor/storage/url-storage.ts

```diff
@@ -53,45 +53,36 @@ export class UrlStorage implements Storage {
     }
   }
 
-  getQueryParam(key: string) {
+  getItem(key: string): string | null {
     return this.getSnappedUrl().searchParams.get(key)
   }
 
-  setPathParam(key: string, value: string) {
-    console.log(`To implement setPathParam key='${key}', value='${value}'`) // TODO: GSLUX-570
-  }
-
-  getPathParam(key: string) {
-    console.log(`To implement getPathParam key='${key}'`)
-    return this.getSnappedUrl().pathname.indexOf(key) // TODO: GSLUX-570
-  }
+  removeItem() {}
 
-  setItem(key: string, value: string, fromPath?: boolean) {
-    if (fromPath) {
-      this.setPathParam(key, value)
-    } else {
-      this.setQueryParam(key, value)
-    }
+  encodeQueryParam(key: string, value: string) {
+    return `${encodeURIComponent(key)}=${encodeURIComponent(value)}`
   }
+}
 
-  getItem(key: string): string | null
-  getItem(key: string, fromPath?: boolean) {
-    let value
+export class UrlStorageAsPath extends UrlStorage {
```

Comment:
Maybe put this class in a separate file?

Reply:
Done.